planning/plan_execution/src/plan_execution.cpp

```diff
@@ -457,6 +457,8 @@ void plan_execution::PlanExecution::planningSceneUpdatedCallback(const planning_
 
 void plan_execution::PlanExecution::doneWithTrajectoryExecution(const moveit_controller_manager::ExecutionStatus &status)
 {
+  // sync all planning scene updates before continuing
+  planning_scene_monitor_->syncSceneUpdates();
```

Comment:
It's important to sync the scene state after execution to allow proper continuation with further planning steps. I'm not sure, whether I catched all places, where planning starts and thus should sync the scene state.